diffcp/cone_program.py

```diff
@@ -131,31 +139,39 @@ def solve_and_derivative(A, b, c, cone_dict, warm_start=None, **kwargs):
     m, n = A.shape
     N = m + n + 1
     cones = cone_lib.parse_cone_dict(cone_dict)
+    cones_parsed = cone_lib.parse_cone_dict_cpp(cones)
     z = (x, y - s, np.array([1]))
     u, v, w = z
+
     Q = sparse.bmat([
         [None, A.T, np.expand_dims(c, - 1)],
         [-A, None, np.expand_dims(b, -1)],
         [-np.expand_dims(c, -1).T, -np.expand_dims(b, -1).T, None]
     ])
-    D_proj_dual_cone = _diffcp.dprojection(
-        v, cone_lib.parse_cone_dict_cpp(cones), True)
+    Q_dense = Q.todense()
+
+    D_proj_dual_cone = _diffcp.dprojection(v, cones_parsed, True)
+    if mode == "dense":
+        M = _diffcp.M_dense(Q_dense, cones_parsed, u, v, w)
+        MT = M.T
+    elif mode == "sparse":
+        M = (Q - sparse.eye(N)) @ dpi_sparse_matrix(z, cones) + sparse.eye(N)
+        MT = M.T
+
+    if mode == "lsqr":
```

Comment:
nit: for symmetry with the above, change to

```
else:
    M = _diffcp.M_operator...
```

Reply:
Done (this was left over from the older dense/sparse mode falling back on `lsqr` for the PSD cone